lib/broadcast-operator.ts

```diff
@@ -252,10 +252,8 @@ export class BroadcastOperator<EmitEvents extends EventsMap, SocketData>
         responses.length === expectedClientCount
       ) {
         clearTimeout(timer);
-        ack.apply(this, [
-          null,
-          this.flags.expectSingleResponse ? null : responses,
-        ]);
+        if (this.flags.expectSingleResponse) ack.apply(this, responses);
```

Comment:
Shouldn't it be `if (this.flags.expectSingleResponse) ack.apply(this, responses[0]);` in this case?

So that:

```js
const [remoteSocket] = await io.in(theSocketId).fetchSockets();

remoteSocket.timeout(1000).emit("hello", (err, response) => {

});

```

Behaves like:

```js
io.on("connection", (socket) => {
  socket.timeout(1000).emit("hello", (err, response) => {

  });
});
```

What do you think?